### PR TITLE
update listApis.json, Quota API, and fixed double url value handling

### PR DESCRIPTION
**Pull request by tonymmm1**
https://cloudstack.apache.org/api/apidocs-4.19/

Quota api implementation

### CODE REVIEW

**Comment by tonymmm1** [comment]
Requesting a review if any mistakes/issues in implementation.

**Comment by tonymmm1** [comment]
I noticed that the listApis.json was missing the api functions.

**Comment by tonymmm1** [comment]
quotaUpdate that was generated has additional response fields in the struct when the documentation states it should only show "updated_on".

```
type QuotaUpdateResponse struct {
	JobID      string `json:"jobid"`
	Jobstatus  int    `json:"jobstatus"`
	Updated_on string `json:"updated_on"`
}
```

**Comment by tonymmm1** [comment]
```user@linux make mocks
user1@s1-desktopt1 go test -v ./...
# github.com/apache/cloudstack-go/v2/cloudstack
cloudstack/QuotaService.go:235:5: v declared and not used
cloudstack/QuotaService.go:241:5: v declared and not used
cloudstack/QuotaService.go:849:5: v declared and not used
cloudstack/QuotaService.go:1418:5: v declared and not used
FAIL    github.com/apache/cloudstack-go/v2/cloudstack [build failed]
FAIL    github.com/apache/cloudstack-go/v2/examples [build failed]
?       github.com/apache/cloudstack-go/v2/generate     [no test files]
FAIL    github.com/apache/cloudstack-go/v2/test [build failed]
FAIL
```

**Comment by tonymmm1** [comment]
seems to be an issue with generator.go on double types

**Comment by tonymmm1** [comment]
Fixed the double/float type generation.

```Running Suite: Encoding Suite - /home/user1/git/cloudstack-go/test
==================================================================
Random Seed: 1721788077

Will run 4 of 4 specs
••••

Ran 4 of 4 Specs in 0.000 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestEncodeValues (0.00s)
=== RUN   TestUUID
--- PASS: TestUUID (0.00s)
PASS
ok      github.com/apache/cloudstack-go/v2/test 0.044s
```

**Comment by tonymmm1** [comment]
As of cloudstack 14.19.0.2 it seems like the quotaBalance response object does not match listApis.json from cmk-cli.


Is the object I am receiving from cloudstack api.

```
{
  "quotabalanceresponse": {
    "balance": {
      "startquota": 0,
      "credits": [],
      "startdate": "2024-07-27T11:32:15-0500",
      "currency": "$"
    }
  }
}
```

Is the generated type.

```
type QuotaBalance struct {
	Account   string  `json:"account"`
	Accountid int64   `json:"accountid"`
	Domain    int64   `json:"domain"`
	JobID     string  `json:"jobid"`
	Jobstatus int     `json:"jobstatus"`
	Name      string  `json:"name"`
	Quota     float64 `json:"quota"`
	Type      int     `json:"type"`
	Unit      string  `json:"unit"`
}
```

There seems to be no correlation between these types?

**Comment by tonymmm1** [comment]
Implemented overrides for quotaBalance and quotaStatement that should work.

File: generate/generate.go [hunk 1]
```diff
@@ -1865,6 +1865,20 @@ func (s *service) generateResponseType(a *API) {
...
+		pn("type QuotaBalanceResponse struct {")
+		pn("    Statement QuotaBalanceResponseType `json:\"balance\"`")
+		pn("}")
+		pn("")
+		pn("type QuotaBalanceResponseType struct {")
+		pn("    StartQuota float64  `json:\"startquota\"`")
+		pn("    Credits    []string `json:\"credits\"`")
+		pn("    StartDate  string   `json:\"startdate\"`")
+		pn("    Currency   string   `json:\"currency\"`")
+		pn("}")
```
Comment (Pearl1594) [inline generate/generate.go:1879]: Do we need to add EndDate and EndQuota as well?

**Review by Pearl1594 — APPROVED** [review]

File: generate/generate.go (hunk 1 above)
Comment (DaanHoogland) [reply generate/generate.go:1879]: @Pearl1594 is this resolved?

**Comment by DaanHoogland** [comment]
@tonymmm1 can you deal with the conflict, please?

**Comment by DaanHoogland** [comment]
@tonymmm1 , I rebased your code in PR #98 . please have a quick review if you can.

**Pull request closed without merging**